app/src/main/java/tech/pauly/findapet/data/models/Animal.kt

```diff
@@ -1,77 +1,159 @@
 package tech.pauly.findapet.data.models
 
+import android.arch.persistence.room.Entity
+import android.arch.persistence.room.Ignore
+import android.arch.persistence.room.PrimaryKey
+import io.reactivex.Single
 import org.simpleframework.xml.Element
 import org.simpleframework.xml.ElementList
 import org.simpleframework.xml.Root
+import tech.pauly.findapet.shared.ResourceProvider
 
-@Root(name = "pet", strict = false)
-open class Animal {
+private const val filenameSuffix = ".png"
+
+sealed class Animal {
 
     @field:Element
     open var id: Int = 0
-        protected set
 
     @field:Element
     open lateinit var shelterId: String
-        protected set
 
     @field:Element
     open lateinit var name: String
-        protected set
+        internal set
 
     @field:Element(name = "animal")
-    private lateinit var _type: String
+    internal lateinit var _type: String
     open val type: AnimalType
         get() = AnimalType.fromString(_type)
 
+    @Ignore
     @field:ElementList(name = "breeds", entry = "breed")
     open lateinit var breedList: List<String>
-        protected set
-
-    open val formattedBreedList: String
-        get() = breedList.joinToString(" / ")
+        internal set
 
     @field:Element
     open lateinit var mix: String
-        protected set
+        internal set
 
     @field:Element(name = "age")
-    private lateinit var _age: String
+    internal lateinit var _age: String
     open val age: Age
         get() = Age.fromString(_age)
 
     @field:Element(name = "sex")
-    private lateinit var _sex: String
+    internal lateinit var _sex: String
     open val sex: Sex
         get() = Sex.fromString(_sex)
 
     @field:Element(name = "size")
-    private lateinit var _size: String
+    internal lateinit var _size: String
     open val size: AnimalSize
         get() = AnimalSize.fromString(_size)
 
     @field:ElementList(name = "options", entry = "option")
-    private lateinit var _options: List<String>
+    internal lateinit var _options: List<String>
     open val options: List<Option>
         get() = _options.map { Option.fromString(it) }
 
-    @field:Element
-    open lateinit var lastUpdate: String
-        protected set
-
+    @Ignore //TODO: Don't ignore https://www.pivotaltracker.com/story/show/157158560
```

Comment:
Have an issue with figuring out how I want to cache Contact info, probably in its own table, but that seems a bit excessive. Going to delay this until we show contact info